make-it/Makefile

```diff
@@ -96,6 +96,12 @@ else
         CC=clang++
         LDFLAGS+= -dead_strip
         CFLAGS+= -pthread $(CCOPT) $(TARGETARCH) -fno-rtti -fno-exceptions -std=c++11
+
+        ifdef ASAN
```

Comment:
It looks like gcc 4.8 supports address sanitizer. Did you try address sanitizer flags with gcc? If that works, I think we should try and leave the flags on for Travis CI builds and see how that goes? If there are performance (build/test run time), then we should leave it on for tag(release) builds.

Reply:
I am going to wait on approving this until we find out if we can enable it for gcc on Travis.

Reply:
I know the flags work with GCC, but I haven't tried it with this project. I should have some time today to give it a shot. It does check for leaks on Linux by default, so I may need to add fixes for those if I add the flags there, FYI.

ASan has significant performance impact: `make testnative` takes about 4x as long (from ~25 seconds to ~100 seconds on my machine). I leave it to you guys to decide if the performance impact is worth it to run with it constantly (outside of released bits).

Reply:
Could probably test it pretty quick by changing:

https://github.com/ni/VireoSDK/blob/da7c8c23213ff5b7590f0a206e9f7a754b4621db/.travis.yml#L47

To `make native ASAN=1`

We probably wouldn't turn it on all the time locally but on the CI (ubuntu trusty gcc4.9) seems fine.

You can try it in a brach, just make sure you have Travis CI enabled in your fork: https://github.com/ni/VireoSDK/blob/master/CONTRIBUTING.md#connect-fork-to-ci-servers

Reply:
Thanks for the suggestion @rajsite! I was able to get a passing Travis CI with ASan enabled: https://travis-ci.org/tchipley/VireoSDK/builds/452022689, so if y'all want it enabled there I can turn it on. I had to change `make unittest` to `make unittest ASAN=1` as well as changing `make native`.

> We probably wouldn't turn it on all the time locally but on Travis CI (ubuntu trusty gcc4.9) seems fine.

> Then, it is not worth it turning it on in Travis CI builds by default. For every commit someone pushes on their PR, we will end up incurring the penalty. Turning it on for tagged(Release) builds is good enough, as bugs of this nature should be rare enough.

@rajsite and @sanmut: I'm getting mixed messages about enabling ASan in Travis CI. I want to make sure everybody knows that ASan is not intended to be enabled on released binaries, which it looks like the Travis CI generates.

Reply:
We can enable it if appropriate in a separate PR. No need to do so in this PR.

You are right that the linux build we make is released. However, there are not any known consumers of that build. It is true that if we publish a linux build we should not have that build be published with ASAN enabled. Maybe we don't really need to publish the Linux build. We can have those discussions in a separate PR.

Could you add a bit of doc to the README for Linux that describes using the ASAN flag https://github.com/ni/VireoSDK/blob/master/README.md#vireo-linux-native-target

After that looks good to me!

Reply:
Cool. I will submit a PR to enable ASan in Travis CI once this is in for discussion.

I added a note to the readme about the ASAN flag. I also noticed the macOS instructions didn't mention that make works there as well, so I added a note that the Linux instructions will work on that platform as well. I also added myself to the author list per your other comment.